api-resources.md

```diff
@@ -0,0 +1,414 @@
+
+`struct Query` is essentially identical to `struct BatchSelector`, so the note about `BatchSelector.fixed_size` from `PUT aggregate_share` applies here, except that a collection's ID is chosen by the collector.
+
+##### DELETE
+
+Instructs the leader to abandon the collection and allows the leader to discard all state related to it. This is akin to DELETE on an aggregate share for the helper (indeed, the leader might send a DELETE to the helper's aggregate share resource after its corresponding collection is deleted).
+
+This is analogous to DAP-02's `DELETE` on a collect job URI.
+
+#### Alternatives: `collection-id`, `PUT` vs. `POST` and idempotence?
+
+This proposal introduces the notion of a `collection-id`, which is chosen by the collector when it sends `CollectReq` to helper. In DAP-02, a collection is already uniquely identified by the combination of the aggregation parameter and the query, meaning we can get by without `collection-id`. Let's sketch out what the API would look like so we can weigh pros and cons.
+
+`POST /tasks/task-id/collections`
+
+The body is a `CollectReq`, defined as above. The request method is `POST` instead of `PUT` because I don't think this request can be idempotent. Let's say a collector makes a time interval query over some interval _i_ and it is received by the leader at time _t1_. At _t1_, the leader has _n1_ reports that fall into the interval _i_. Then, suppose at least one more report that falls into _i_ arrives and the leader and helper prepare it. Then, at _t2 > t1_, the collector sends a query with the same interval _i_ again.
+
+Should the leader serve up the results it computed at time _t1_? Or should it make a new collection, consuming another unit of max batch query, that includes the reports that arrived and were prepared between _t1_ and _t2_?
+
+If collections are identified by `collection-id`, then there's no ambiguity: the collector can poll `GET /tasks/task-id/collections/collection-id` to obtain the same collection over and over again. `PUT /tasks/task-id/collections/collection-id` with the same `collection-id` is also unambiguous, and if the collector wants to make a new query with the same interval, it can choose a new `collection-id` and `PUT` that, which the leader will service if the task's parameters allow it.
+
+The cons of `collection-id` is that an aggregation parameter and a `struct Query` don't uniquely identify a collection. So if a collector wants to find out if there's an existing collect job that meets its parameter, it has to enumerate all of them using `GET /tasks/task-id/collections`. We could improve this either with [HTTP QUERY](https://www.ietf.org/archive/id/draft-ietf-httpbis-safe-method-w-body-02.html) or query params on `GET /tasks/task-id/collections`.
+
+### Collections
+
+Only the leader supports this resource.
+
+#### Path
+
+`/tasks/task-id/collections`
+
+#### Representation
+
+A paginated list of collections for the task.
```

Comment:
Why do we need a paginated list of collection jobs?

I think this can be dropped from the specification (& left up to implementations, if they find it handy).

Reply:
Deleted with the other list resources